pydantic_evals/pydantic_evals/datasets.py

```diff
@@ -0,0 +1,376 @@
+            except ValueError as e:
+                if isinstance(e.__cause__, ValidationError):
+                    raise ValueError(
+                        f'{cls.__name__} dataset file {dataset_path} already exists, but does not contain compatible data.'
+                        f' Fix or delete the file before calling this function:\n{e.__cause__}'
+                    ) from e.__cause__
+                else:
+                    raise
+            dataset_text = dataset_path.read_text()
+            cases_text_with_schema = _ensure_yaml_language_server_line(dataset_text, schema_ref)
+            if cases_text_with_schema != dataset_text:
+                dataset_path.write_text(cases_text_with_schema)
+        else:
+            content = yaml.dump(to_jsonable_python(cls(rows=[])), sort_keys=False)
+            dataset_path.write_text(f'{yaml_language_server_line}\n{content}')
+        return schema_ref
+
+    @classmethod
+    def model_json_schema_with_assessments(cls) -> dict[str, Any]:
+        registry = cls.assessment_registry()
+
+        assessment_types: list[Any] = []
+        for name, function in registry.items():
+            signature = inspect.signature(function)
+            first_param = list(signature.parameters.values())[0]
+            type_hints = _typing_extra.get_function_type_hints(function)
+            type_hints.pop(first_param.name)
+            type_hints.pop('return', None)
+            required_type_hints: dict[str, Any] = {}
+
+            for p in signature.parameters.values():
+                if p.name == first_param.name:
+                    continue
+                if p.name not in type_hints:
+                    type_hints[p.name] = Any
+                if p.default is not p.empty:
+                    type_hints[p.name] = NotRequired[type_hints[p.name]]
+                else:
+                    required_type_hints[p.name] = type_hints[p.name]
+
+            if len(type_hints) == 0:
+                assessment_types.append(Literal[name])
+            elif len(type_hints) == 1:
+                type_hint_type = next(iter(type_hints.values()))  # pyright: ignore
+                td = TypedDict(f'short_assessment_{name}', {name: type_hint_type})  # pyright: ignore
+                td.__pydantic_config__ = {'extra': 'forbid'}  # pyright: ignore
+                assessment_types.append(td)
+            else:
+                if len(required_type_hints) == 1 and len(type_hints) > 1:
+                    # Short option: only one required parameter, so we can drop the nesting
+                    type_hint_type = next(iter(required_type_hints.values()))  # pyright: ignore
+                    td = TypedDict(f'short_assessment_{name}', {name: type_hint_type})  # pyright: ignore
+                    td.__pydantic_config__ = {'extra': 'forbid'}  # pyright: ignore
+                    assessment_types.append(td)
+
+                # Long form: multiple parameters, or multiple required parameters
```

Comment:
this long form should be allowed in all cases

Reply:
Should we also support the `call: ...; args: [...]; kwargs: {...}` format I just added, for everything?

Happy to try it and see, I don't want to implement that just yet though because there's enough other stuff to do still..